tests/test_execution.py

```diff
@@ -1,26 +1,18 @@
-import substra
-
 import pytest
+import copy
+
+import substra
 
 import substratest as sbt
 
 
-def test_tuples_execution_on_same_node(factory, session):
+def test_tuples_execution_on_same_node(data_network):
     """Execution of a traintuple, a following testtuple and a following traintuple."""
-    spec = factory.create_dataset()
-    dataset = session.add_dataset(spec)
+    factory, network = data_network
+    session = copy.deepcopy(network.sessions[0])
```

Comment:
What about adding a `copy()` method to the `Session` class?

Reply:
It'd make things a bit easier, why not!